moxin-protocol/src/open_ai.rs

```diff
@@ -0,0 +1,133 @@
+use std::collections::HashMap;
+use serde::{Deserialize, Serialize};
+use crate::data::*;
+
+// Based on https://platform.openai.com/docs/api-reference/chat/object
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ChatRequestData {
+    // Not really necesary but it is part of the OpenAI API. We are going to send the id
+    // of the model currently loaded.
+    pub model: ModelID,
+
+    pub frequency_penalty: Option<f32>,
+    pub logprobs: Option<bool>,
+    pub top_logprobs: Option<u32>,
+    pub max_tokens: Option<u32>,
+    pub presence_penalty: Option<f32>,
+    pub seed: Option<u32>,
+    pub stop: Option<Vec<String>>,
+    pub stream: Option<bool>,
+    pub temperature: Option<f32>,
+    pub top_p: Option<f32>,
+
+    // Adding the following fields since there are part of the OpenAI API,
+    // but are not likely to be used in the first version of the client
+    pub n: Option<u32>,
+    pub logit_bias: Option<HashMap<String, f32>>,
+}
```

Comment:
This request is incorrect, it's missing user input. It seems to be a response from OpenAI，not a request.

Reply:
Correct, it's missing the `messages` field, the first one defined here https://platform.openai.com/docs/api-reference/chat/object. It's the most important part of the request for sure!

Adding it now.

Note that the rest of the fields are actually part of the request (not the response), according to this OpenAI definition.

Reply:
Done! Thanks for catching this issue.